adapters/facebook_test.go

```diff
@@ -285,6 +307,220 @@ func TestFacebookBasicResponse(t *testing.T) {
 	}
 }
 
+func TestFacebookInterstitialResponse(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(DummyFacebookServer))
+	defer server.Close()
+
+	fbdata = bidInfo{
+		partnerID:   12345678,
+		domain:      "nytimes.com",
+		page:        "https://www.nytimes.com/2017/05/04/movies/guardians-of-the-galaxy-2-review-chris-pratt.html?hpw&rref=movies&action=click&pgtype=Homepage&module=well-region&region=bottom-well&WT.nav=bottom-well&_r=0",
+		publisherID: "987654321",
+		tags:        make([]tagInfo, 1),
+		deviceIP:    "25.91.96.36",
+		deviceUA:    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_4) AppleWebKit/603.1.30 (KHTML, like Gecko) Version/10.1 Safari/603.1.30",
+		buyerUID:    "need-an-actual-fb-id",
+	}
+	fbdata.tags[0] = tagInfo{
+		code:        "first-tag",
+		placementID: fmt.Sprintf("%s_999998888", fbdata.publisherID),
+		bid:         1.67,
+		W:           300,
+		H:           250,
+	}
+
+	conf := *DefaultHTTPAdapterConfig
+	an := NewFacebookAdapter(&conf, fmt.Sprintf("%d", fbdata.partnerID), "localhost")
+	an.URI = server.URL
+	an.nonSecureUri = server.URL
+
+	pbin := pbs.PBSRequest{
+		AdUnits: make([]pbs.AdUnit, 2),
+	}
+	for i, tag := range fbdata.tags {
+		pbin.AdUnits[i] = pbs.AdUnit{
+			Code:       tag.code,
+			MediaTypes: []string{"BANNER"},
+			Sizes: []openrtb.Format{
+				{
+					W: tag.W,
+					H: tag.H,
+				},
+			},
+			Bids: []pbs.Bids{
+				pbs.Bids{
+					BidderCode: "audienceNetwork",
+					BidID:      fmt.Sprintf("random-id-from-pbjs-%d", i),
+					Params:     json.RawMessage(fmt.Sprintf("{\"placementId\": \"%s\"}", tag.placementID)),
+				},
+			},
+			Instl: 1,
+		}
+	}
+
+	body := new(bytes.Buffer)
+	err := json.NewEncoder(body).Encode(pbin)
+	if err != nil {
+		t.Fatalf("Json encoding failed: %v", err)
+	}
+
+	fmt.Println("body", body)
+
+	req := httptest.NewRequest("POST", server.URL, body)
+	req.Header.Add("Referer", fbdata.page)
+	req.Header.Add("User-Agent", fbdata.deviceUA)
+	req.Header.Add("X-Real-IP", fbdata.deviceIP)
+
+	pc := pbs.ParsePBSCookieFromRequest(req)
+	pc.TrySync("audienceNetwork", fbdata.buyerUID)
+	fakewriter := httptest.NewRecorder()
+	pc.SetCookieOnResponse(fakewriter, "")
+	req.Header.Add("Cookie", fakewriter.Header().Get("Set-Cookie"))
+
+	cacheClient, _ := dummycache.New()
+	pbReq, err := pbs.ParsePBSRequest(req, cacheClient)
+
+	ctx := context.TODO()
```

Comment:
`context.TODO()`, or `context.Background()`?

godocs mention that [the latter is generally used in tests](https://golang.org/pkg/context/#Background)